src/tir/usmp/analysis/extract_buffer_info.cc

```diff
@@ -505,6 +572,41 @@ BufferInfoAnalysis BufferInfoExtractor::operator()(const PrimFunc& main_func) {
       open_set.erase(le_event.buffer_info);
     }
   }
+
+  // All ConstantPoolInfo items should have conflicts with each other
+  // as they will be placed in RO segment and pre-initialized
+
+  // split buffers to vars (WorkspacePoolInfo items) and constants (ConstantPoolInfo items)
+  Array<BufferInfo> buffer_info_vars;
+  Array<BufferInfo> buffer_info_constants;
+  for (const auto& kv : this->buffer_info_map_) {
+    const auto& stmt = kv.second;
+    if (stmt->IsInstance<AllocateConstNode>()) {
+      buffer_info_constants.push_back(kv.first);
+    } else {
+      buffer_info_vars.push_back(kv.first);
+    }
+  }
+  ICHECK(buffer_info_map_.size() == buffer_info_vars.size() + buffer_info_constants.size())
+      << "missing value";
+
+  Map<ObjectRef, ObjectRef> srch;
+  // intersect with each other, as all constants should exist at the same time
```

Comment:
did this comment get split from previous one?

Reply:
no. this is an attempt to describe logic of the code in steps.

Reply:
ok, i'm not sure the sentence reads well to me...mind revising the comment a bit?

Reply:
revised (a bit).